src/ContainerRegistry/ContainerRegistry/Models/ContainerRegistryDataPlaneClient.cs

```diff
@@ -100,6 +107,12 @@ public void SetEndPoint(string RegistryName)
         {
             _endPoint = RegistryName.ToLower() + '.' + _suffix;
             _client.LoginUri = _https + _endPoint;
+            _credential = new Track2TokenCredential(new DataServiceCredential(AzureSession.Instance.AuthenticationFactory,
+                        _context, AzureEnvironment.ExtendedEndpoint.ContainerRegistryEndpointResourceId));
+            _client2 = new Track2.ContainerRegistryClient(new Uri(_https + _endPoint), _credential, new Track2.ContainerRegistryClientOptions()
+            {
+                Audience = _context.Environment.ExtendedProperties[AzureEnvironment.ExtendedEndpoint.ContainerRegistryEndpointResourceId]
```

Comment:
credential has already contained resource id of ACR. Shall it be added again as one of client options?

Reply:
I've seen that the definition of Track2 Client requires both credential and audience.